apis/gearexe/src/api/api.ts

```diff
@@ -27,4 +34,20 @@ export class GearExeApi {
   get provider() {
     return this._provider;
   }
+
+  public setSigner(signer: ISigner): this {
+    this._signer = signer;
+    return this;
+  }
```

Comment:
Wouldn't be better to pass it to a constructor instead of using standalone method?